src/server/shell/decoration/basic_manager.h

```diff
@@ -18,34 +18,56 @@
 #define MIR_SHELL_DECORATION_BASIC_MANAGER_H_
 
 #include "manager.h"
+
+#include <mir/observer_registrar.h>
+
 #include <functional>
 #include <unordered_map>
 #include <mutex>
 
+namespace mir::graphics { class DisplayConfigurationObserver; }
+
 namespace mir
 {
 class Executor;
 namespace graphics
 {
 class GraphicBufferAllocator;
+class DisplayConfiguration;
 }
 namespace shell
 {
 class Shell;
 namespace decoration
 {
 class Decoration;
+class DisplayConfigurationListener;
+
+class OutputListener
+{
+public:
+    virtual void advise_output_update(mir::graphics::DisplayConfiguration const& config) = 0;
+
+protected:
+    OutputListener() = default;
+    virtual ~OutputListener() = default;
+    OutputListener(OutputListener const&) = delete;
+    OutputListener operator=(OutputListener const&) = delete;
+};
```

Comment:
I see no reason to make `OutputListener` part of the `basic_manager.h` interface. Or even why we need that interface.

`OutputListener` is only used by `msd::DisplayConfigurationListener` which is implemented entirely in `basic_manager.cpp` and that just needs something to call (e.g. `std::function<void(float new_scale)>`)